### PR TITLE
Fixed a problem with travasing ancestors.

### PR DESCRIPTION
**Pull request by shiraji**
see #1591

### CODE REVIEW

File: AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/menu/OptionsMenuItemTest.java [hunk 1]
```diff
@@ -36,4 +36,10 @@ public void testWhenFieldNameCouldConflictWithParameterName() throws IOException
 		assertCompilationSuccessful(result);
 	}
 
+	@Test
+	public void testDeeplyInheritedMenuNotBreak() throws IOException {
+		CompileResult result = compileFiles(OptionsMenuItemSubSubSubActivity.class);
+		assertCompilationSuccessful(result);
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/menu/OptionsMenuItemTest.java]: Is this test sufficient? You should check actually the code is in order, right?

Comment (shiraji) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/menu/OptionsMenuItemTest.java]: Yes. You are right. I fixed the test case.

**Comment by shiraji** [comment]
mmm...I couldn't reproduce this failure. I ran following commands:

```shell
mvn clean
mvn install -Dandroid.sdk.path=/opt/android-sdk/
```

Could you point how I can fix this problem, @WonderCsabo ?

**Comment by shiraji** [comment]
```java
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        this.menu = menu.findItem(R.id.menu);
        this.menu1 = menu.findItem(R.id.menu1);
        this.menu2 = menu.findItem(R.id.menu2);
        this.menu3 = menu.findItem(R.id.menu3);
        return super.onCreateOptionsMenu(menu);
    }
```

There is `R.id`...mmm, let me create another test cases.

**Comment by WonderCsabo** [comment]
We are using a better way to generate imports to R classes, so we no longer
import the inner R classes, but only R. This is the recommended on Android.
See #1548

**Comment by shiraji** [comment]
@WonderCsabo Could you give me a sample unit test code? After I learn phax/jcodemodel, I still don't understand how I implement this unit test code. Thanks.

**Comment by WonderCsabo** [comment]
@shiraji i just rebased your branch to latest develop use the latest modifications for jcodemodel, and fixed the test case. Check out [this](https://github.com/WonderCsabo/androidannotations/tree/orderAncestors) branch.

**Comment by WonderCsabo** [comment]
BTW @shiraji you do not need to log the generated Java code. You can just check the file under *target/generated-test*.

**Comment by shiraji** [comment]
Thanks, let me fix the issue.

I logged the generated Java code just because I want to know what is the difference between my local generated code and travic ci's generated code.

**Comment by WonderCsabo** [comment]
Please squash the last commit into commit before.

**Comment by shiraji** [comment]
Here you go.

**Pull request merged**